server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -58,6 +64,21 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
     HashMap<String, String> perspectiveMap = analyzeWithPerspective(cumulativeComments);
     NLPResult commentsSentiment = analyzeWithNLP(cumulativeComments);
 
+    try {
+      AnalysisCache cache = new AnalysisCache();
+      cache.open();
+      cache.add(
+          "example key",
+          new AnalysisGroup(perspectiveMap, commentsSentiment, commentArray, videoId, videoInfo));
+      cache.close();
+    } catch (InvalidKeyException
+        | IllegalBlockSizeException
+        | BadPaddingException
+        | NoSuchAlgorithmException
+        | NoSuchPaddingException e) {
+      e.printStackTrace();
+    }
+
```

Comment:
This code definitely works, but it seems like it would become increasingly slow if you do an I/O operation that rewrites the whole file on every request. What are your plans here?

Reply:
I had two thoughts:
  1. Rather than rewrite the file, I can just append the latest entry and delete any existing, stale matches.
  2. Store all the cached data that the user gets in that session, and then when the session ends rewrite the entire file.
  3. Do both.

Reply:
Consider making your cache persist in memory between web requests. I'm a bit hazy with how AppEngine handles lifecycles for your app, but I think this would be possible and keep your user from waiting on file reads and periodically flushing your cache to the filesystem.